compiler/damlc/tests/daml-test-files/Asset.daml

```diff
@@ -0,0 +1,13 @@
+daml 1.2
+module Asset where
+
+import DA.Record
+
+-- These are some simple typeclasses that can be used as preconditions for
+-- generic templates, for example in the Fungible and Transferrable templates.
+-- It would be better to write these as constraint synonyms, but DAML doesn't
+-- allow them at the moment.
+
+class (Eq t, HasField "issuer" t Party, HasField "owner" t Party) => Asset t
```

Comment:
Not relevant here since this is just a test but once we start moving towards a library for generic templates (docs are somewhere in the middle), I’m not a fan of relying on `HasField` instances here. E.g. consider your test below where you only have self-issued assets. You can’t make a type just for those since you force users to have both `issuer` and `owner` fields separately.
A more compelling example would be a type where `amount` can be derived from other quantities but you force users to have a redundant field that can get out of sync.

Reply:
Yeah, I agree these type classes are too restrictive at the moment. We'll need to find a good interface for our generic template library.